school.c

```diff
@@ -0,0 +1,743 @@
+// Function to create a new student node
+Student* createStudentNode(const char* firstName, const char* lastName, const char* phoneNumber, int grade, int class, int marks[]) {
+    Student* newStudent = (Student*)malloc(sizeof(Student));
+    if (newStudent != NULL) {
+        strcpy(newStudent->firstName, firstName);
+        strcpy(newStudent->lastName, lastName);
+        strcpy(newStudent->phoneNumber, phoneNumber);
+        newStudent->grade = grade;
+        newStudent->class = class;
+        for (int i = 0; i < 10; ++i) {
+            newStudent->marks[i] = marks[i];
+        }
+        newStudent->next = NULL;
+    }
+    return newStudent;
+}
+
+
+// Function to add a student to the school
+void addStudent(Student* newStudent) {
+    if (school[newStudent->grade-1][newStudent->class-1] == NULL) {
+        // If no student in class - add as first student
+        school[newStudent->grade-1][newStudent->class-1] = newStudent;
+    } else {
+        // Add student to the head of the list
+        Student* currentStudent = school[newStudent->grade-1][newStudent->class-1];
+        school[newStudent->grade-1][newStudent->class-1] = newStudent;
+        newStudent->next = currentStudent;
+    }
+}
+
+
+// Function to find a student by first name and last name in the school system
+Student* findStudent(const char* firstName, const char* lastName) {
+    for (int grade = 0; grade < NUM_GRADES; ++grade) {
+        for (int classNum = 0; classNum < NUM_CLASSES; ++classNum) {
+            Student* currentStudent = school[grade][classNum];
+            while (currentStudent != NULL) {
+                if (strcmp(currentStudent->firstName, firstName) == 0 && strcmp(currentStudent->lastName, lastName) == 0) {
+                    return currentStudent;
+                }
+                currentStudent = currentStudent->next;
+            }
+        }
+    }
+    return NULL; // Student not found
+}
+
+
+// Function to delete a student from the school system
+void deleteStudent() {
+    char firstName[50], lastName[50];
+
+    printf("Enter the first name: ");
+    get_input(firstName, 50);
+    printf("Enter the last name: ");
+    get_input(lastName, 50);
+
+    // Find the student in the school system
+    Student* targetStudent = findStudent(firstName, lastName);
+
+    if (targetStudent == NULL) {
+        printf("Student not found.\n");
+        return;
+    }
+
+    printf("Found student:\n");
+    printf("Name: %s %s\n", targetStudent->firstName, targetStudent->lastName);
+    printf("Grade: %d\n", targetStudent->grade);
+    printf("Class: %d\n", targetStudent->class);
+    printf("Phone Number: %s\n", targetStudent->phoneNumber);
+    printf("\n");
+
+    // Ask the user to confirm the deletion
+    printf("Do you want to delete this student? (y/n): ");
+    char confirm, c;
+    scanf(" %c", &confirm);
+    c = getchar();
+    while (c != '\n') c = getchar();
+
+
+    if (confirm == 'y' || confirm == 'Y') {
+        // Delete the student
```

Comment:
You have already found the student earlier, so why search for him again?
If you need the prev student, save it while searching for the student above.